hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/OperativeEndpoint.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.fs.azurebfs.services;
+
+import org.apache.hadoop.fs.azurebfs.AbfsConfiguration;
+
+/**
+ * This class is mainly to unify the fallback for all API's to DFS endpoint at a single spot.
+ */
+public class OperativeEndpoint {
+        public static boolean isMkdirEnabledOnDFS(PrefixMode mode, AbfsConfiguration abfsConfiguration, boolean isMarkerBlob) {
+            if (mode == PrefixMode.BLOB) {
+                boolean configEnabled = abfsConfiguration.shouldMkdirFallbackToDfs();
+                return configEnabled && isMarkerBlob;
+            }  else return mode == PrefixMode.DFS;
```

Comment:
since, Blob and DFS can be the only DFS. lets do :
```
if() {} else {
 return true;
}
```

Reply:
Done